tracing-syslog/src/syslog.rs

```diff
@@ -0,0 +1,220 @@
+    /// Reserved for local use.
+    LOG_LOCAL7 = libc::LOG_LOCAL7,
+}
+
+impl Default for Facility {
+    fn default() -> Self {
+        Self::LOG_USER
+    }
+}
+
+/// `syslog` severity.
+#[allow(non_camel_case_types)]
+#[derive(Copy, Clone)]
+#[repr(i32)]
+pub enum Severity {
+    /// System is unusable.
+    LOG_EMERG = libc::LOG_EMERG,
+    /// Action must be taken immediately.
+    LOG_ALERT = libc::LOG_ALERT,
+    /// Critical conditions.
+    LOG_CRIT = libc::LOG_CRIT,
+    /// Error conditions.
+    LOG_ERR = libc::LOG_ERR,
+    /// Warning conditions.
+    LOG_WARNING = libc::LOG_WARNING,
+    /// Normal, but significant, condition.
+    LOG_NOTICE = libc::LOG_NOTICE,
+    /// Informational message.
+    LOG_INFO = libc::LOG_INFO,
+    /// Debug-level message.
+    LOG_DEBUG = libc::LOG_DEBUG,
+}
+
+impl From<Level> for Severity {
+    fn from(level: Level) -> Self {
+        match level {
+            Level::ERROR => Self::LOG_ERR,
+            Level::WARN => Self::LOG_WARNING,
+            Level::INFO => Self::LOG_NOTICE,
+            Level::DEBUG => Self::LOG_INFO,
+            Level::TRACE => Self::LOG_DEBUG,
+        }
+    }
+}
+
+/// `syslog` priority.
+#[derive(Copy, Clone, Debug)]
+struct Priority(libc::c_int);
+
+impl Priority {
+    fn new(facility: Facility, level: Level) -> Self {
+        let severity = Severity::from(level);
+        Self((facility as libc::c_int) | (severity as libc::c_int))
+    }
+}
+
+fn syslog(priority: Priority, msg: &CStr) {
+    unsafe { libc::syslog(priority.0, "%s\0".as_ptr().cast(), msg.as_ptr()) }
+}
+
+/// Creates a `syslog`-backed [`tracing_core::Collect`].
+pub fn logger(
+    identity: impl Into<Cow<'static, CStr>>,
+    options: Options,
+    facility: Facility,
+) -> impl Collect {
+    Registry::default().with(Syslog::new(identity, options, facility))
+}
+
+/// [`tracing_subscriber::Subscribe`] that logs to `syslog`.
+pub struct Syslog {
+    /// Identity e.g. program name. Referenced by syslog, so we store it here to
+    /// ensure it lives until we are done logging.
+    #[allow(dead_code)]
+    identity: Cow<'static, CStr>,
+    facility: Facility,
+}
+
+impl Syslog {
+    /// Creates a new `syslog` logger.
+    pub fn new(
+        identity: impl Into<Cow<'static, CStr>>,
+        options: Options,
+        facility: Facility,
+    ) -> Self {
+        let identity = identity.into();
+        // SAFETY: identity will remain alive until the returned struct's fields
+        // are dropped, by which point `closelog` will have been called by the
+        // `Drop` implementation.
+        unsafe { libc::openlog(identity.as_ptr(), options.0, facility as libc::c_int) };
+        Syslog { identity, facility }
+    }
+}
+
+impl Drop for Syslog {
+    fn drop(&mut self) {
+        unsafe { libc::closelog() };
+    }
+}
+
+impl<C> Subscribe<C> for Syslog
+where
+    C: Collect + for<'span> LookupSpan<'span>,
+{
+    fn on_event(&self, event: &Event, _ctx: Context<C>) {
```

Comment:
It doesn't seem like you're even _using_ the `Registry` here. I'd suggest implementing the [`new_span` method](https://tracing.rs/tracing_subscriber/subscribe/trait.subscribe#method.new_span) in addition to `on_event`, as it'll allow you collect the span's fields. Take a look at how `FmtSubscriber` is implemented: https://github.com/tokio-rs/tracing/blob/master/tracing-subscriber/src/fmt/fmt_subscriber.rs#L545

Reply:
This is now outdated with the `MakeWriter`-based approach